db/migrations/20230202110100-add-repo-sync-state-fk.js

```diff
@@ -0,0 +1,20 @@
+"use strict";
+
+module.exports = {
+	up: async (queryInterface) => {
+		await queryInterface.addConstraint("RepoSyncStates", ["subscriptionId"], {
+			type: "foreign key",
+			name: "fk_reposyncst_subs_subid",
+			references: {
+				table: "Subscriptions",
+				field: "id"
+			},
+			onUpdate: "cascade",
+			onDelete: "cascade"
```

Comment:
Instead of fix codes in the apps, adding tests to make sure RepoSyncStates are deleted when we uninstall the Subscriptions everywhere, I just use `cascade`

Reply:
Can you make sure that cascade doesn't actually delete the subscription?  Finding documentation on this to show exactly what cascade does isn't great

Reply:
I believe the `cascade`, in this case, should be `if subscription is deleted, all related repo sync state will also be deleted`

Reply:
I can add some tests though.

Reply:
Okay, tests added.

Reply:
> Finding documentation on this to show exactly what cascade does isn't great...

I use the doc here https://sequelize.org/v5/class/lib/query-interface.js~queryinterface#instance-method-addConstraint

And this is the screenshot of what actual constraint is created.

![image](https://user-images.githubusercontent.com/105693507/217155457-b93be4b5-cabf-40f1-9d77-3763adc1ac6e.png)